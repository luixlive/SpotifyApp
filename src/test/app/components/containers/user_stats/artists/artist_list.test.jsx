import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import {
  artistList as componentProps,
} from './../../../../../test_utils/components_props';
import initialState from './../../../../../test_utils/initial_state';
import { mockTopArtists } from './../../../../../test_utils/mock_data';
import ArtistList, {
  PureArtistList,
/* eslint-disable-next-line max-len */
} from './../../../../../../app/components/containers/user_stats/artists/artist_list';

describe('App Components - Artist List', () => {
  describe('Snapshots', () => {
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
    });

    it('renders with some artists', () => {
      props.artists = _.cloneDeep(mockTopArtists);
      const rendered = renderer.create(<PureArtistList {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders while reloading', () => {
      props.reloading = true;
      const rendered = renderer.create(<PureArtistList {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeAll(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <ArtistList />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureArtistList).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureArtistList).props()))
        .toEqual(Object.keys(componentProps));
    });
  });
});
