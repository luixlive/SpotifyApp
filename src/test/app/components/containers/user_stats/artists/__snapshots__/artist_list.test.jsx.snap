// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - Artist List Snapshots renders while reloading 1`] = `
<div
  className="ui basic segment screen-loader"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted text loader"
        >
          <h3
            className="ui header"
          >
            
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components - Artist List Snapshots renders with error 1`] = `
<div
  className="ui center aligned container"
>
  <h1
    className="ui header"
    style={
      Object {
        "fontSize": "4em",
        "marginTop": "1.5em",
      }
    }
  >
    Error :(
  </h1>
  <h2
    className="ui header"
  >
    Some error
  </h2>
</div>
`;

exports[`App Components - Artist List Snapshots renders with some artists 1`] = `
<div
  className="ui basic segment"
>
  <div
    className="ui grid"
  >
    <div
      className="eight wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          1
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name1
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="eight wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          2
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name2
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="eight wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          3
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name3
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components - Artist List Snapshots renders with some artists mobile 1`] = `
<div
  className="ui basic segment"
>
  <div
    className="ui grid"
  >
    <div
      className="sixteen wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          1
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name1
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="sixteen wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          2
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name2
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
    <div
      className="sixteen wide column"
    >
      <div>
        <h2
          className="ui left floated header"
        >
          3
        </h2>
        <div
          className="ui fluid card artist"
          href={undefined}
          onClick={[Function]}
        >
          <a
            href="spotify.url"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              className="ui image"
              src="image.url"
            />
          </a>
          <div
            className="content"
          >
            <div
              className="header"
            >
              <h3
                className="ui header"
              >
                <a
                  href="spotify.url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  name3
                </a>
              </h3>
            </div>
            <div
              className="meta"
            >
              <h4
                className="sub header"
              />
            </div>
          </div>
          <div
            className="extra content"
          >
            Followers: 
            0
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
